src/gt4py/next/iterator/type_system/inference.py

```diff
@@ -584,11 +567,19 @@ def visit_FunCall(
                 return ts.DeferredType(constraint=None)
             assert isinstance(tuple_.type, ts.TupleType)
             return tuple_.type.types[index]
+        shifts = {}
```

Comment:
The code quality should be improved. Please document also the motivation for this special behavior in a comment.